src/java.base/share/classes/java/time/ZoneOffset.java

```diff
@@ -423,12 +426,12 @@ public static ZoneOffset ofTotalSeconds(int totalSeconds) {
         if (totalSeconds < -MAX_SECONDS || totalSeconds > MAX_SECONDS) {
             throw new DateTimeException("Zone offset not in valid range: -18:00 to +18:00");
         }
-        if (totalSeconds % (15 * SECONDS_PER_MINUTE) == 0) {
-            Integer totalSecs = totalSeconds;
-            ZoneOffset result = SECONDS_CACHE.get(totalSecs);
+        int quarters = totalSeconds / SECONDS_PER_QUARTER;
+        if (totalSeconds - quarters * SECONDS_PER_QUARTER == 0) {
+            ZoneOffset result = QUARTER_CACHE.getOpaque(quarters & 0xff);
```

Comment:
For the ease of future maintenance, I recommend replacing `quarters & 0xFF` with a new variable `key`:
```java
// quarters range from -72 to 72.
// & 0xFF maps them to 0-72 and 184-255.
int key = quarters & 0xFF;
```